File: 3581-the-two-sneaky-numbers-of-digitville/3581-the-two-sneaky-numbers-of-digitville.java
class Solution {
    public int[] getSneakyNumbers(int[] nums) {
        int n = nums.length;
        int[] hash = new int[102];
        int[] ans = new int[2];
        int idx = 0;
        for (int i = 0; i < n; ++i) {
            hash[nums[i]]++;
            if (hash[nums[i]] == 2)
                ans[idx++] = nums[i];
        }
        return ans;
    }
}

File: 3581-the-two-sneaky-numbers-of-digitville/3581-the-two-sneaky-numbers-of-digitville.go
func getSneakyNumbers(nums []int) []int {
    n := len(nums)
    hash := make([]int, 102)
    ans := make([]int, 2)
    idx := 0
    for i := 0; i < n; i++ {
        hash[nums[i]]++
        if hash[nums[i]] == 2 {
            ans[idx] = nums[i]
            idx += 1
        }
    }
    return ans
}